### PR TITLE
Migrate run make cdylib

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of https://github.com/rust-lang/rust/issues/121876.

r? @jieyouxu

### CODE REVIEW

File: tests/run-make/cdylib/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// When the cdylib crate type was added as a variation of dylib, it needed a test to check its
+// function. See <https://github.com/rust-lang/rust/pull/33553>.
```
Comment (jieyouxu) [inline tests/run-make/cdylib/rmake.rs:2]: Suggestion: this comment isn't very useful. Maybe we can improve its wording, like:

This test tries to check that basic cdylib libraries can be compiled and linked successfully with C code, that the cdylib itself can depend on another rlib, and that the library can be built with LTO.

- `bar.rs` is a rlib
- `foo.rs` is a cdylib that relies on an extern crate `bar` and defines two `extern "C"` functions:
    - `foo()` which calls `bar::bar()`.
    - `bar()` which implements basic addition.

**Review by jieyouxu — APPROVED** [review]
Thanks! Looks good in general, just left some suggestions.

File: tests/run-make/cdylib/rmake.rs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+use std::fs::remove_file;
+
+use run_make_support::{dynamic_lib, is_msvc, run, rustc, tmp_dir, Cc};
+
+fn main() {
+    rustc().input("bar.rs").run();
+    rustc().input("foo.rs").run();
+
+    if is_msvc() {
+        Cc::new()
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply tests/run-make/cdylib/rmake.rs:15]: :open_mouth:

File: tests/run-make/cdylib/rmake.rs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+// with C code, that the cdylib itself can depend on another rlib, and that the library can be built
+// with LTO.
+//
+// - `bar.rs` is a rlib
+// - `foo.rs` is a cdylib that relies on an extern crate `bar` and defines two `extern "C"`
+//   functions:
+//     - `foo()` which calls `bar::bar()`.
+//     - `bar()` which implements basic addition.
+
+// @ignore-cross-compile
```
Comment (jieyouxu) [inline tests/run-make/cdylib/rmake.rs:11]: Problem: this needs to be

```suggestion
//@ ignore-cross-compile
```

**Review by jieyouxu — CHANGES_REQUESTED** [review]
Thanks, just one problem

**Comment by jieyouxu** [comment]
r=me after fixing the directive :3

File: tests/run-make/cdylib/rmake.rs (hunk 3 above)
Comment (jieyouxu) [reply tests/run-make/cdylib/rmake.rs:11]: cc https://github.com/rust-lang/rust/issues/125813

**Review by jieyouxu — APPROVED** [review]

**Comment by jhpratt** [comment]
@bors r-

needs formatting https://github.com/rust-lang/rust/pull/125839#issuecomment-2143295179

**Comment by GuillaumeGomez** [comment]
Fixed formatting.

**Pull request merged**